### PR TITLE
Allow cipher strings to be given using its standard name

### PR DESCRIPTION
**Pull request by eriklax**
Added pull request for issue #16169

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: doc/man1/openssl-ciphers.pod.in [hunk 1]
```diff
@@ -115,6 +115,8 @@ used. The format is described below.
 The cipher list consists of one or more I<cipher strings> separated by colons.
 Commas or spaces are also acceptable separators but colons are normally used.
 
+The cipher string may reference a cipher using its standard name.
```
Comment (richsalz) [inline doc/man1/openssl-ciphers.pod.in:118]: You should have a reference to *where* the standard names are listed, such as https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-4

Comment (eriklax) [reply doc/man1/openssl-ciphers.pod.in]: Good point! Updated docs

**Comment by mattcaswell** [comment]
OTC: This is fine for master, but cannot be backported to 3.0 or 1.1.1

**Comment by t8m** [comment]
Could you please rebase to fix the conflict in CHANGES.md?

**Comment by eriklax** [comment]
> Could you please rebase to fix the conflict in CHANGES.md?

Done!

File: CHANGES.md [hunk 2]
```diff
@@ -41,6 +41,11 @@ breaking changes, and mappings for the large list of deprecated functions.
 
 ### Changes between 1.1.1 and 3.0 [xx XXX xxxx]
 
+ * The SSL_CTX_set_cipher_list family functions now accept ciphers using their
+   standard names.
```
[the remark this answers is not in the document]
Comment (mattcaswell) [reply CHANGES.md:45]: Perhaps modify this wording to say "IANA standard names"?

File: ssl/ssl_ciph.c [hunk 3]
```diff
@@ -1095,6 +1095,10 @@ static int ssl_cipher_process_rulestr(const char *rule_str,
                     && (ca_list[j]->name[buflen] == '\0')) {
                     found = 1;
                     break;
+                } else if (ca_list[j]->stdname && strncmp(buf, ca_list[j]->stdname, buflen) == 0
```
Comment (mattcaswell) [inline ssl/ssl_ciph.c:1098]: Please change `ca_list[j]->stdname` to `ca_list[j]->stdname != NULL`

File: CHANGES.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (eriklax) [reply CHANGES.md]: Fixed

File: ssl/ssl_ciph.c (hunk 3 above)
Comment (eriklax) [reply ssl/ssl_ciph.c]: Fixed

**Review by paulidale — APPROVED** [review]

File: ssl/ssl_ciph.c [hunk 4]
```diff
@@ -1095,6 +1095,10 @@ static int ssl_cipher_process_rulestr(const char *rule_str,
                     && (ca_list[j]->name[buflen] == '\0')) {
                     found = 1;
                     break;
+                } else if (ca_list[j]->stdname != NULL && strncmp(buf, ca_list[j]->stdname, buflen) == 0
+                    && (ca_list[j]->stdname[buflen] == '\0')) {
```
Comment (t8m) [inline ssl/ssl_ciph.c:1099]: Formatting and indentation nit. The code should look like:
```
                } else if (ca_list[j]->stdname != NULL
                           && strncmp(buf, ca_list[j]->stdname, buflen) == 0
                           && ca_list[j]->stdname[buflen] == '\0') {
```

Comment (eriklax) [reply ssl/ssl_ciph.c]: Looks much better, fixed

**Comment by t8m** [comment]
I now realized there is a CLA: trivial on the first commit. Could you please remove that via `git rebase -i master` and force push it?

**Comment by eriklax** [comment]
> I now realized there is a CLA: trivial on the first commit. Could you please remove that via `git rebase -i master` and force push it?

Done, did it turn out ok?

**Comment by paulidale** [comment]
I've not seen your ICLA.  When did you send it?

**Comment by t8m** [comment]
@eriklax Please follow the instructions at https://www.openssl.org/policies/cla.html to send the ICLA.

**Comment by eriklax** [comment]
> @eriklax Please follow the instructions at https://www.openssl.org/policies/cla.html to send the ICLA.

- [x] ICLA sent by email

**Review by t8m — APPROVED** [review]

**Comment by mattcaswell** [comment]
The CI failures in the "minimal" builds look relevant.

**Comment by t8m** [comment]
Also please rebase to fix the CHANGES.md conflict.

**Comment by eriklax** [comment]
> The CI failures in the "minimal" builds look relevant.

Changed cipher in test

> Also please rebase to fix the CHANGES.md conflict.

Done

**Review by paulidale — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master branch. Thank you for your contribution.

**Pull request closed without merging**